lightning/src/offers/invoice_error.rs

```diff
@@ -17,27 +17,27 @@ use crate::util::string::UntrustedString;
 
 use crate::prelude::*;
 
-/// An error in response to an [`InvoiceRequest`] or an [`Invoice`].
+/// An error in response to an [`InvoiceRequest`] or an [`Bolt12Invoice`].
```

Comment:
What if we just renamed `Bolt11Invoice`s and left `Bolt12Invoice` as `Invoice`? That way we don't have ambiguity with `InvoiceRequest`

Reply:
Would prefer to just qualify both `Invoice` types. `InvoiceRequest` is sorta borderline for me, but I'm leaning to not renaming it. There's no name clash there and it's a higher-level type rather than being an auxiliary type for `Invoice`.